env/env_basic_test.cc

```diff
@@ -17,15 +17,81 @@
 #include "test_util/testharness.h"
 
 namespace ROCKSDB_NAMESPACE {
+typedef Env* CreateEnvFunc();
```

Comment:
The changes here to avoid the early initialization before main is ok to me. I think it is ready to merge.